sdk/search/azure-search/src/test/java/com/azure/search/SearchIndexClientTestBase.java

```diff
@@ -92,22 +92,20 @@
     }
 
     SearchIndexClientBuilder getClientBuilder(String indexName) {
-        if (!interceptorManager.isPlaybackMode()) {
-            return new SearchIndexClientBuilder()
-                .endpoint(endpoint)
-                .indexName(indexName)
-                .httpClient(new NettyAsyncHttpClientBuilder().wiretap(true).build())
-                .credential(searchApiKeyCredential)
-                .addPolicy(interceptorManager.getRecordPolicy())
-                .addPolicy(new RetryPolicy())
-                .addPolicy(new HttpLoggingPolicy(
-                    new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS)));
-        } else {
-            return new SearchIndexClientBuilder()
-                .endpoint(endpoint)
-                .indexName(indexName)
-                .httpClient(interceptorManager.getPlaybackClient());
+        SearchIndexClientBuilder builder = new SearchIndexClientBuilder().endpoint(endpoint)
+            .indexName(indexName);
+        if (interceptorManager.isPlaybackMode()) {
+            return builder.httpClient(interceptorManager.getPlaybackClient());
+        }
+        builder.httpClient(new NettyAsyncHttpClientBuilder().wiretap(true).build())
+            .credential(searchApiKeyCredential)
+            .addPolicy(new RetryPolicy())
+            .addPolicy(new HttpLoggingPolicy(
+                new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS)));
```

Comment:
_Not this PR and for documentation purposes._

This change would ideally be done by setting the `HttpLogOptions` within the builder. Right now this will cause the generated pipeline to have two logging policies which is less than ideal when running live tests with logging on.